tests/ui/impl-trait/in-trait/issue-102140.stderr

```diff
@@ -20,14 +20,6 @@ LL |         MyTrait::foo(&self)
    |
    = help: the trait `MyTrait` is implemented for `Outer`
 
-error[E0277]: the trait bound `&dyn MyTrait: MyTrait` is not satisfied
-  --> $DIR/issue-102140.rs:23:9
-   |
-LL |         MyTrait::foo(&self)
```

Comment:
Why do the 3 diagnostics have different spans? Is there a way to have the standard diagnostic deduplication catch them by having the same span for the 3 diagnostics?

Reply:
But they're three different errors. You can't return a unsized type from a function, you can't bind it to a variable and you can't have an unsized expression(?). All three can exist independently.